### PR TITLE
Revoke Trust

### PR DESCRIPTION
**Pull request by pogopaule**
Closes https://github.com/yunity/karrot-frontend/issues/2341

## What does this PR do?

Trust can now be revoked.

image

image

image

## Links to related issues

[PR on the backend side](https://github.com/yunity/karrot-backend/pull/1133)

## Checklist

- [ ] added a test, or explain why one is not needed/possible...
- [x] no unrelated changes
- [x] asked someone for a code review
- [x] joined [chat.foodsaving.world/channel/karrot-dev](https://chat.foodsaving.world/channel/karrot-dev)
- [x] added an entry to CHANGELOG.md (description, pull request link, username(s))
- [x] tried out on a mobile device (does everything work as expected on a smaller screen?)

### CODE REVIEW

File: src/group/api/groups.js [hunk 1]
```diff
@@ -72,6 +72,10 @@ export default {
   trustUser (groupId, userId) {
     return axios.post(`/api/groups/${groupId}/users/${userId}/trust/`)
   },
+
+  revokeTrust (groupId, userId) {
+    return axios.post(`/api/groups/${groupId}/users/${userId}/revoketrust/`)
```
Comment (nicksellen) [inline src/group/api/groups.js:77]: A `DELETE `/api/groups/${groupId}/users/${userId}/trust/` API call would seem more consistent with our other API endpoints (e.g. https://github.com/yunity/karrot-frontend/blob/5c6c6e8f2b845c230e6878f249bd6c2ba37efff8/src/messages/api/reactions.js#L21-L23) - which could be described as "_vaguely RESTful_".

File: src/users/components/TrustButton.vue [hunk 2]
```diff
@@ -62,6 +62,18 @@
               width="20px"
             >
           </QBtn>
+          <QBtn
+            v-if="trusted && !user.isCurrentUser"
+            color="negative"
+            :loading="membership.trustUserStatus.pending"
```
Comment (nicksellen) [inline src/users/components/TrustButton.vue:68]: should this be `revokeTrustStatus`? (if that has been wired up...)

**Review by nicksellen — COMMENTED** [review]
Great, thanks for working on this :star: 

A left some comments to invite input, will head over to the backend PR now :)

File: src/group/api/groups.js (hunk 1 above)
Comment (pogopaule) [reply src/group/api/groups.js:77]: Oh yes very good point!

File: src/users/components/TrustButton.vue (hunk 2 above)
Comment (pogopaule) [reply src/users/components/TrustButton.vue:68]: Yes, fixed it. Now the loading spinner works.

**Comment by pogopaule** [comment]
@nicksellen I have not been able to write a working test for the dialog. I don't see much benefit in writing tests for the other lines that are marked as uncovered. What do you think? Merge anyway?

**Comment by nicksellen** [comment]
Yup, I agree @pogopaule.

**Comment by pogopaule** [comment]
> Yup, I agree @pogopaule.

@nicksellen Cool! Could you approve the PR?

**Review by nicksellen — APPROVED** [review]
I think this one slipped through my attention, sorry about that!

It looks good to me, and works as expected :star: 

There is one interesting side effect - the person that created a group has the editor role initially without having received any trust (editor + 0 trust), later when more trust is required to get the editor role, you can remove that initial user by trusting them then revoking the trust.

That's not a side effect of this frontend implementation, but just something I noticed whilst trying it out :)

... this got out of date from master branch, so could do with a rebase, but good to go otherwise.

**Comment by nicksellen** [comment]
I just rebased on top of current master.

**Comment by pogopaule** [comment]
Hey, thanks for rebasing and merging! I have been off the radar for a while, but my plan is to work on karrot again in the weeks to come. :)

**Comment by nicksellen** [comment]
> Hey, thanks for rebasing and merging! I have been off the radar for a while, but my plan is to work on karrot again in the weeks to come. :)

We'll have you any time :+1:  Whenever your rhythm of life comes into sync with Karrot again :heart:

**Pull request merged**